### PR TITLE
added date tests, modified DateSubject for testing

### LINKED ISSUE #39: US8: Advancing Date
Put an additional button in your UI. This button will advance the time by one day, i.e., 24 hours. We may click this button multiple times to advance multiple days.

### PR DESCRIPTION
**Pull request by pink10000**
# Abstract

This Pull Request introduces critical enhancements to the application's date management capabilities.

- **Problem Solved**: The PR addresses the issue of insufficient testing and potential bugs in handling dates within the application. 
- **Solution Approach**:
  - Modification of `DateSubject` to include a null check for `sharedPreferences`, preventing null pointer exceptions.
  - Addition of a new testing suite `DateSubjectTest` to test the functionality of `DateSubject`, including setting, getting, and advancing dates.
- **Changes Made**:
  - **Modified Classes**: `SuccessoratorDatabase` and `DateSubject` for enhanced functionality and reliability.
  - **New Classes**: `DateSubjectTest` introduces testing for date operations.

# Issues

Closes #39 

## Type of Change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Testing

The PR includes extensive unit testing for the date management functionality. Tests were conducted to validate the correct setting and getting of dates, as well as the advancement of dates to ensure accuracy.

# Checklist

- [x] **The code is understandable.**
- [x] **The code is well structured.**
- [x] **The code is correct.**

# Pending Items

Further integration testing may be required to ensure the date management enhancements work seamlessly with other components of the application.

### CODE REVIEW

**Comment by pink10000** [comment]
attempt fix for a major bug

File: lib/build.gradle.kts
Comment (newracket) [inline lib/build.gradle.kts:1]: Undo these changes

**Review by newracket — CHANGES_REQUESTED** [review]

**Review by newracket — APPROVED** [review]
LGTM

**Review by 3dcantaloupe — APPROVED** [review]
lookkign sho good guysss like the tests that you added for testinfg the dates
